README.md

```diff
@@ -140,10 +145,193 @@ class MyEntity
+    
+    /**
+     * @ORM\OneToMany(targetEntity="Parameter", cascade={"persist"})
+     */
+    private $parameters;
+
+    // ...
+}
+```
+
+The `Parameter` entity must implement the following:
+```php
+class Parameter
+{
+    /**
+     * @ORM\ManyToOne(targetEntity="Task")
+     */
+    private $owner;
+    
+    /**
+     * @ORM\Column(type="string")
+     */
+    private $name;
+    
+    /**
+     * @ORM\Column(type="string")
+     * @AG\Generate()
+     */
+    private $value;
+    
+    // This signature is a requirement for enum accessor generation.
+    public function __construct($task, $name, $value)
+    {
+        $this->owner  = $task;
+        $this->>name  = $name;
+        $this->>value = $value;
+    }
+}
+```
+
+### Enum class
+
+The "enum class" only consists of public constants that use a prefix in their names to denote the data type of the
+values they hold in the database.
+
+The following types are supported:
+
+| prefix | type    | example |
+|--------|---------|---------|
+| `S_`   | string  | "foobar"|
+| `I_`   | integer | 1234    |
+| `F_`   | float   | 3.14    |
+| `B_`   | boolean | true    |
+| `A_`   | array   | array   |
+
+Now, lets take the following example for an enum class with some parameters:
+
+```php
+class MyTaskParamNames
+{
+    /**
+     * Represents the client if the task is currently runnnig for.
+     */
+    public const I_CLIENT_ID = 'CLIENT_ID';
+    
+    /**
+     * An awesome URL.
+     */
+    public const S_AWESOME_URL = 'https://www.hostnet.nl/';
+}
+```
+
+Now that we have our three classes (`Task`, `Parameter` and `MyTaskParamNames`), we can start generating code.
+
+### The "Enumerator" annotation
+
+With version 2.8.0 comes the `Enumerator` annotation which can be used inside the existing `Generate` annotation.
+
+Taking the code that we just wrote in the examples above, we can generate an accessor method for `MyTaskParamNames`
+by modifying the annotation of the `parameters` property of our `Task` class.
+
+```php
+class Task
+{
+    use Generated\TaskMethodsTrait;
+
+    /**
+     * @ORM\OneToMany(targetEntity="Parameter", cascade={"persist"})
+     * @AG\Generate(enumerators={
+     *     @AG\Enumerator("MyTaskParamNames", name="MyParams")
+     * })
+     */
+    private $property;
+}
+```
+
+Once the code is generated, you will now have a newly generated class called `MyTaskParamNamesEnum` in the 
+`Generated` directory (and namespace) relative to the namespace of `MyTaskParamNames`. An accessor for this class is
+generated using the `name` property in the `TaskMethodsTrait`.
+
+The accessor for this enum based on the code above will be called `getMyParams()`. You can give this any name you want
+as long as it is suitable for a method name.
+
+Once the code is generated, you'll have access to 5 methods per parameter:
+```php
+
+$task = new Task();
+
+// hasClientId() will check if a parameter with the name I_CLIENT_ID exists in the collection of parameters belonging to
```

Comment:
Break line

Reply:
I don't get it?

It's one line and consists 120 characters. (Look at the line numbers in the gutter)